### PR TITLE
Chris/support image types

### PR DESCRIPTION
**Pull request by sihrc**
URL -> PNG, GIF, JPG, etc

### CODE REVIEW

File: lib/image.js [hunk 1]
```diff
@@ -1,78 +1,115 @@
...
+          throw new Error(
+            "Data image URL input is an invalid image or url: " + image
+          )
         }
+      });
+    } else if (typeof image === 'string') {
+      if (image.length <= 260 && fs.lstatSync(image).isFile()) {
+        input = image;
+        var arr = image.split('.');
+        options = arr[arr.length - 1];
```
Comment (anneCarlson) [inline lib/image.js]: I'd rename this variable `image_type` for clarity

File: lib/image.js [hunk 2]
```diff
@@ -1,78 +1,115 @@
...
-                "Data input type must be array, filepath, or base64 encoded string"
-            );
+          throw new Error(
+            "Data image URL input is an invalid image or url: " + image
+          )
         }
+      });
+    } else if (typeof image === 'string') {
+      if (image.length <= 260 && fs.lstatSync(image).isFile()) {
+        input = image;
```
Comment (anneCarlson) [inline lib/image.js]: I'd rename this variable `image_file` for clarity.

File: test/integration/imagetypes.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  });
+
+  it('PNG Base64', function (done) {
+      image.preprocess(read("base64.txt"), 64, true, false).then(function (result) {
+          result.should.be.a('string');
+          done();
+      });
+  });
+  it('URL JPG', function (done) {
+      image.preprocess("http://www.myhoundhaven.org/images/Golden%20Retriever.jpg", 64, true, false).then(function (result) {
```
Comment (anneCarlson) [inline test/integration/imagetypes.js]: It probably would be better to use an image we host so we can ensure it stays up, but that's a nit.

**Comment by anneCarlson** [comment]
A few comments but overall looks good! Probably worth running ti by someone who has node experience, though!

File: test/integration/imagetypes.js (hunk 3 above)
Comment (madisonmay) [inline test/integration/imagetypes.js]: Even just a quick upload to imgur or something would work here.

Comment (madisonmay) [inline test/integration/imagetypes.js]: http://i.imgur.com/Mfw8Lnd.jpg

Comment (anneCarlson) [inline test/integration/imagetypes.js]: Imgur deletes images that aren't viewed over six months, do we have any idea if this would count as a view?

**Pull request merged**